Artifacts/windows-install-certificate/certificate-install.ps1

```diff
@@ -0,0 +1,109 @@
+<#
+Installs an existing certificate to the LocalMachine store.
+Creates a self signed certificate and imports it into you Personal and Root stores. 
+I used this when setting up a new development site on dev machine.
+#>
+[CmdletBinding()]
+Param(
+    [ValidateNotNullOrEmpty()]
+    [string] $certificateName,
+    [ValidateNotNullOrEmpty()]
+    [string] $base64cert,
+    [ValidateNotNullOrEmpty()]
+    [string] $certificatePassword
+)
+
+
+##################################################################################################
+
+#
+# Powershell Configurations
+#
+
+# Note: Because the $ErrorActionPreference is "Stop", this script will stop on first failure.  
+#       This is necessary to ensure we capture errors inside the try-catch-finally block.
+$ErrorActionPreference = "Stop"
+
+# Ensure we set the working directory to that of the script.
+pushd $PSScriptRoot
+
+# Ensure that current process can run scripts. 
+Set-ExecutionPolicy -ExecutionPolicy Bypass -Scope Process -Force 
+
+#
+# Functions used in this script.
+#
+
+function Handle-LastError
+{
+    [CmdletBinding()]
+    param(
+    )
+
+    $message = $error[0].Exception.Message
+    if ($message)
+    {
+        Write-Host -Object "ERROR: $message" -ForegroundColor Red
+    }
+    
+    # IMPORTANT NOTE: Throwing a terminating error (using $ErrorActionPreference = "Stop") still
+    # returns exit code zero from the PowerShell script when using -File. The workaround is to
+    # NOT use -File when calling this script and leverage the try-catch-finally block and return
+    # a non-zero exit code from the catch block.
+    exit -1
+}
+
+#
+# Handle all errors in this script.
+#
+
+trap
+{
+    # NOTE: This trap will handle all errors. There should be no need to use a catch below in this
+    #       script, unless you want to ignore a specific error.
+    Handle-LastError
+}
+
+###################################################################################################
+
+#
+# Main execution block.
+#
+
+try
+{
+    Write-Host "Installing certificate $certificateName"
+
+    If(-NOT ([Security.Principal.WindowsPrincipal] [Security.Principal.WindowsIdentity]::GetCurrent()).IsInRole([Security.Principal.WindowsBuiltInRole] "Administrator"))
+    {
+        #build up the deploy arguments
+        $arguments = "-file `"{0}`"" -f $script:MyInvocation.MyCommand.Path
+    
+        # Start the new process
+        Start-Process powershell.exe -Verb runas -ArgumentList $arguments
+        exit
```

Comment:
Do you really need to call `exit` here? The call to `Write-Host 'Done'` at the end will not execute.

Also, should we consider adding `-Wait` here? Otherwise, the process will alway be launched but you won't get any feedback if it fails. So, the artifact will be set to `Succeeded` potentially prematurely.

Reply:
This part of the script is not specific to this artifact. It can be copied from any of the MS provided artifacts and I don't want to change the model.

Reply:
Implemented in the latest commit.